nautobot/core/templates/generic/object_list.html

```diff
@@ -163,7 +195,38 @@ <h1>{% block title %}{{ content_type.model_class|meta:"verbose_name_plural"|bett
         // By removing all name attr with 'form-' prefix form input/select elements
         // only filterable fields would be picked up by `$(this).serialize()`
         $(this).find(`input[name*="form-"], select[name*="form-"]`).removeAttr("name")
-        location.replace("?" + $(this).serialize())
+
+        let query_params = $(this).serialize()
+
+        // check if `?q=` or `&q=` in query_params
+        if(location.search && /(\?|\&)q/.test(location.search)){
+            // Append `q` value to `query_params`
+            query_params += "&q=" + /(?<=(\?|\&)q=)\w+/.exec(location.search)[0]
+        }
+        location.replace("?" + query_params)
```

Comment:
This is confusing to me, at least based on the comments... if q is in the query_params, we append it to the query_params?

Reply:
updated the comment: check if `?q=` or `&q=` in  url query params